core-java-modules/core-java-uuid/src/main/java/com/baeldung/uuid/UUIDLongGenerator.java

```diff
@@ -0,0 +1,58 @@
+package com.baeldung.uuid;
+
+import java.nio.ByteBuffer;
+import java.util.UUID;
+
+public class UUIDLongGenerator {
```

Comment:
You have found that the probability of collisions with these approaches is between 50% and 100% by experimentation. The title of the article is about generating a unique positive long, so I think it's important that we do our best to generate a positive value, since the negative rate is so high.

So how about, instead of allowing for negative values, this could be a `UUIDPositiveLongGenerator` and you could simply call `Math.abs` on each of the results to ensure a positive value? The test can then be changed to assert that all the values generated are positive.

Then the statistics you generate in the test can just focus on collisions.